Bug: YAML float version (e.g. "1.1") causes silent persona load failure with empty error message

## Summary

When a portfolio element's frontmatter has a version field with only two components (e.g. `version: 1.1`), YAML parses it as a JavaScript float rather than a string. This causes a silent load failure with an empty error message:

```
[ERROR] Failed to load persona from /Users/mick/.dollhouse/portfolio/personas/mcp-qa-engineer.md:
```

Note the blank message after the colon — a TypeError from calling a string method (`.match()`, `.split()`) on a number produces an error with no message in this context.

## Steps to Reproduce

1. Create a persona with `version: 1.1` (or any two-component version) in the frontmatter — no quotes
2. Start the server
3. Observe the silent `[ERROR] Failed to load persona from ...:` with no further detail

## Root Cause

YAML spec treats bare `1.1` as a float. Two-component versions like `1.0`, `1.1`, `2.0` are all affected. Three-component versions like `1.1.0` are strings and parse correctly.

The fix was partially applied:

- ✅ `MetadataService.ts:185` — does `String(normalized.version)` before normalizing
- ❌ `BaseElement.ts:400` — second assignment with no coercion:

```typescript
// Current (broken for floats):
this.version = parsed.version || '1.0.0';

// Fix:
this.version = String(parsed.version || '1.0.0');
```

When persona loading takes the code path through line 400 instead of `normalizeMetadata`, the `String()` coercion is bypassed and the float reaches string-method calls downstream.

## Workaround

Use three-component versions in frontmatter: `version: 1.1.0` instead of `version: 1.1`. Quoting also works: `version: "1.1"`.

## Fix

Add `String()` wrap at `BaseElement.ts:400`:

```typescript
this.version = String(parsed.version || '1.0.0');
```

May be worth auditing for other unguarded `parsed.version` assignments as well.

## Labels

`bug`, `good first issue`

--- a/src/elements/BaseElement.ts
+++ b/src/elements/BaseElement.ts
@@ -397,7 +397,7 @@ export abstract class BaseElement implements IElement {
       // Update properties
       this.id = parsed.id;
       this.type = parsed.type;
-      this.version = parsed.version || '1.0.0';
+      this.version = normalizeVersion(String(parsed.version ?? '1.0.0'));
       this.metadata = parsed.metadata;
       this.references = parsed.references || [];
       this.extensions = parsed.extensions || {};

--- a/src/elements/memories/Memory.ts
+++ b/src/elements/memories/Memory.ts
@@ -13,7 +13,7 @@
  * 6. Audit logging for all operations
  */
 
-import { BaseElement } from '../BaseElement.js';
+import { BaseElement, normalizeVersion } from '../BaseElement.js';
 import { IElement, ElementValidationResult, ValidationError } from '../../types/elements/index.js';
 import { ElementType } from '../../portfolio/types.js';
 import { IElementMetadata } from '../../types/elements/IElement.js';
@@ -938,7 +938,7 @@ export class Memory extends BaseElement implements IElement {
 
       // Update properties
       this.id = parsed.id;
-      this.version = parsed.version || '1.0.0';
+      this.version = normalizeVersion(String(parsed.version ?? '1.0.0'));
       this.metadata = parsed.metadata || {};
       this.extensions = parsed.extensions || {};
 

--- a/src/elements/skills/Skill.ts
+++ b/src/elements/skills/Skill.ts
@@ -10,7 +10,7 @@
  * 5. MEDIUM: Added XSS protection through input sanitization
  */
 
-import { BaseElement } from '../BaseElement.js';
+import { BaseElement, normalizeVersion } from '../BaseElement.js';
 import { IElement, IElementMetadata, ElementValidationResult } from '../../types/elements/index.js';
 import { ElementType } from '../../portfolio/types.js';
 import { logger } from '../../utils/logger.js';
@@ -450,7 +450,7 @@ export class Skill extends BaseElement implements IElement {
       
       // Update ID and version if provided
       if (parsed.id) this.id = parsed.id;
-      if (parsed.version) this.version = parsed.version;
+      if (parsed.version != null) this.version = normalizeVersion(String(parsed.version));
       
       // Restore parameters
       if (parsed.parameters) {

--- a/src/elements/templates/Template.ts
+++ b/src/elements/templates/Template.ts
@@ -11,7 +11,7 @@
  * 6. MEDIUM: Unicode normalization to prevent homograph attacks
  */
 
-import { BaseElement } from '../BaseElement.js';
+import { BaseElement, normalizeVersion } from '../BaseElement.js';
 import { IElement, IElementMetadata, ElementValidationResult } from '../../types/elements/index.js';
 import { ElementType } from '../../portfolio/types.js';
 import { logger } from '../../utils/logger.js';
@@ -957,7 +957,7 @@ export class Template extends BaseElement implements IElement {
       
       // Update ID and version if provided
       if (parsed.id) this.id = parsed.id;
-      if (parsed.version) this.version = parsed.version;
+      if (parsed.version != null) this.version = normalizeVersion(String(parsed.version));
       
       // Clear compiled template and section caches (content changed)
       this.compiledTemplate = undefined;

--- a/tests/unit/elements/BaseElement.test.ts
+++ b/tests/unit/elements/BaseElement.test.ts
@@ -5,10 +5,10 @@
 import { BaseElement, normalizeVersion } from '../../../src/elements/BaseElement.js';
 import { ElementType } from '../../../src/portfolio/types.js';
 import { ElementStatus } from '../../../src/types/elements/index.js';
-import { createTestMetadataService } from '../../helpers/di-mocks.js';
+import { MetadataService } from '../../../src/services/MetadataService.js';
 
 // Create a shared MetadataService instance for all tests
-const metadataService = createTestMetadataService();
+const metadataService = new MetadataService();
 
 // Create a concrete implementation for testing
 class TestElement extends BaseElement {
@@ -261,6 +261,28 @@ describe('BaseElement', () => {
       expect(element.id).toBe(original.id);
       expect(element.metadata.name).toBe('Original');
       expect(element.metadata.description).toBe('Original description');
+    });
+
+    it('should coerce numeric versions during deserialization', () => {
+      const original = new TestElement({
+        name: 'Original',
+        description: 'Original description'
+      });
+
+      const json = JSON.stringify({
+        id: original.id,
+        type: original.type,
+        version: 1.1,
+        metadata: original.metadata,
+        references: original.references,
+        extensions: original.extensions,
+        ratings: original.ratings
+      });
+
+      const element = new TestElement();
+      element.deserialize(json);
+
+      expect(element.version).toBe('1.1.0');
     });
     
     it('should throw on invalid JSON', () => {

